src/views/challenges/challenge.js

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect, useState } from "react";
+import { Link } from "@reach/router";
+import { Chrome, Button } from "components/components";
+import { createRecord } from "db/records";
+
+export const Challenge = ({ challengeId, location: { state } }) => {
+  const [records, setRecords] = useState([]);
+  useEffect(() => {
+    const { records } = state;
+    records && setRecords(records);
+  }, [state]);
+  return (
+    <Chrome
+      title={`Challenge ${challengeId}`}
+      links={{
+        left: {
+          to: "..",
+          text: "Back",
+        },
+        right: {
+          to: "settings",
+          text: "Edit",
+        },
+      }}
+    >
+      <Button
+        onClick={async () => {
+          const record = {
+            challengeId,
+            timestamp: new Date().getTime(),
+            value: 1,
+          };
+          await createRecord(record).then((id) => {
+            setRecords([...records, { ...record, id }]);
+          });
+        }}
+      >
+        Add new record with value 1
+      </Button>
+      {records.map(({ id, value }) => (
```

Comment:
For some reason if you navigate to the edit challenge or edit record route and go back, it doesn't render any of the records. When you navigate here from the challenges view though, it renders them just fine.

Reply:
Nice catch!

**Reason for problem:**
Records data is passed from `Challenges` view to `Challenge` view via `@reach/router`'s location state in order to minimize unnecessary calls to the database.

**Problem:**
Records data is not passed from the `Challenge` view to the `EditChallenge` nor the `EditRecord` views since it is not needed there.

**Solution:**
Upon arriving in the `Challenge` view, a check is made to see if records data is contained in the location state. If so, it means we're coming from `Challenges` and the data is used. If not, it means we're coming from somewhere else and therefore we make a call to the database in order to populate the records data.

Cheers! ☕